doc/hotspot-style.md

```diff
@@ -696,6 +696,15 @@ The HotSpot macro `ATTRIBUTE_ALIGNED` provides similar capabilities for
 platforms that define it. This macro predates the use by HotSpot of C++
 versions providing `alignas`. New code should use `alignas`.
 
+### alignof
+
+`alignof`
+([n2341](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2007/n2341.pdf))
+is permitted alongside `alignas` for checking alignment as a less verbose
+alternative to `std::alignment_of<>`. The template precedes the operator
+in the language, so `alignof` should be preferred to `std::alignment_of<>`
+if possible.
```

Comment:
`alignof` doesn't "check" anything. And this is still overly verbose. Keep it
simple. I was going to suggest using this:
"`alignof`(ref...) is permitted.  Use `alignof` instead of `std::alignment_of<>`."

But I think better is to add `alignof` to the "### Additional Permitted
Features" list, and change the place where inclusion of `<type_traits>` is
discussed, adding a restriction against using `std::alignment_of<>`. There one
can justify the restriction by noting the trait class preceded the operator
and provides the same functionality. We're probably going to want additional
restrictions for `<type_traits>` because of
https://bugs.openjdk.org/browse/JDK-8298399.

So edit that section like so:

* `#include <type_traits>` with restrictions (see below).
...

Some definitions from `<type_traits>` should not be used.
* `std::alignment_of<>` has been superseded by `alignof`, which should be used instead.

Reply:
Will do, thanks